### DISCUSSION TITLE
How do I send a verification email when extending with go?

### DISCUSSION THREAD

**Discussion by Drive4Code** [pocketbase/pocketbase#6146] [Q&A] [answered]
I'm trying to send a verification email from the collection's default template from extending with go (pb 0.23.10), but can't seem to figure it out. I have an api endpoint to create a new user where I manually create the user's record, so i have access to the `*core.RequestEvent`. It would also be fine for me to understand how to implement it through a Hook `OnRecordAfterCreateSuccess`, if that's easier to explain

**Comment by ganigeorgiev** [comment, marked as answer]
I'll consider documenting it better, but for now you can use the [`mails` subpackage helpers](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.23.10/mails), something like:

```go
app.OnRecordAfterCreateSuccess("users").BindFunc(func(e *core.RecordEvent) error {
	if !e.Record.Verified() {
		err := mails.SendRecordVerification(e.App, e.Record)
		if err != nil {
			return apis.NewBadRequestError("The user was created successfully but failed to send verification email", err)
		}
	}

	return e.Next()
})
```

**Reply by Drive4Code** [reply]
Thank you!

**Reply by ganigeorgiev** [reply]
_Edit: updated the example to add check for the record verified state before sending the email because the user could be created by superuser or programmatically with preset verified state._
